iceoryx_posh/source/popo/trigger.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/popo/trigger.hpp"
+#include "iceoryx_posh/internal/popo/building_blocks/condition_variable_signaler.hpp"
+
+namespace iox
+{
+namespace popo
+{
+Trigger::~Trigger()
+{
+    reset();
+}
+
+bool Trigger::hasTriggered() const noexcept
+{
+    return (isValid() && m_hasTriggeredCallback) ? m_hasTriggeredCallback().value() : false;
```

Comment:
can this race? Which method could run in parallel in different threads? When I see this code I hope the answer is "nothing is allowed to run in parallel"

Reply:
This is actually the contract. Nothing is allowed to run in parallel but I will state this explicitly in the readme.